homeassistant/components/caldav/config_flow.py

```diff
@@ -0,0 +1,175 @@
+            except CALDAV_EXCEPTIONS:
+                errors["base"] = "cannot_connect"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                return self.async_create_entry(
+                    title=f"{url.host} ({user_input[CONF_USERNAME]})",
+                    data=user_input,
+                    options={
+                        CONF_CALENDARS: user_input.get(CONF_CALENDARS, []),
+                        CONF_CUSTOM_CALENDARS: user_input.get(
+                            CONF_CUSTOM_CALENDARS, []
+                        ),
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_custom_calendars(self, user_input=None) -> FlowResult:
+        """Add custom calendars."""
+        if user_input.get(CONF_ADD_CUSTOM_CALENDAR):
+            user_input[CONF_ADD_CUSTOM_CALENDAR] = False
+            self.user_input = user_input
+        elif user_input:
+            self.user_input.update(user_input)
+            return await self.async_step_user(self.user_input)
+
+        return self.async_show_form(
+            step_id="custom_calendars", data_schema=CUSTOM_CALENDARS_SCHEMA
+        )
+
+    async def async_step_import(self, import_config: dict[str, str]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+
+class CaldavOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle option."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Init object."""
+        self.config_entry = config_entry
+        self.calendars = config_entry.options.get(CONF_CALENDARS, [])
+        self.custom_calendars = config_entry.options.get(CONF_CUSTOM_CALENDARS, [])
+
+    async def async_step_init(self, user_input=None) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        errors = {}
+        options_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_CALENDARS, default=self.calendars
+                ): selector.SelectSelector(
+                    selector.SelectSelectorConfig(
+                        options=[], custom_value=True, multiple=True
+                    ),
+                ),
+                vol.Optional(
+                    CONF_CUSTOM_CALENDARS,
+                    default=[],
+                    description={"suggested_value": self.custom_calendars},
+                ): selector.ObjectSelector(),
+            }
+        )
+        if user_input:
+            if user_input[CONF_CALENDARS] and user_input[CONF_CUSTOM_CALENDARS]:
+                errors["base"] = "choice_custom"
+            else:
+                return self.async_create_entry(title="", data=user_input)
```

Comment:
You could reverse this condition, swap the branches so it returns first which would eliminate the `else` and then outdent